BUG: Fix formatting error in exception message

aleju opened the pull request:
This fixes a simple formatting error in an exception message in `random/common.pyx`. The message is supposed to contain the expected vs. the actually received dtype, but instead contained two times the expected dtype. This would lead to confusing error messages such as `Supplied output array has the wrong type. Expected <class 'numpy.float64'>, got <class 'numpy.float64'>`.

Example code to enforce the error message:
```python
import numpy as np
arr = np.zeros((1,), dtype="float32")
gen = np.random.Generator(np.random.SFC64(0))
gen.random(size=(1,), dtype="float64", out=arr)
```
Output after the fix:
```
TypeError: Supplied output array has the wrong type. Expected <class 'numpy.float64'>, got float32
```

aleju committed:
BUG: Fix formatting error in exception message

This commit fixes a simple formatting error in the
generation of an exception message. The message
is supposed to contain the expected vs. the actual
dtype, but instead contained two times the expected
dtype.

seberg commented on numpy/random/common.pyx:
```diff
@@ -227,7 +227,7 @@ cdef check_output(object out, object dtype, object size):
     if out_array.dtype != dtype:
         raise TypeError('Supplied output array has the wrong type. '
-                        'Expected {0}, got {0}'.format(dtype, out_array.dtype))
+                        'Expected {0}, got {1}'.format(dtype, out_array.dtype))
```
Thanks. It probably makes sense to call `dtype = np.dtype(dtype)` for nicer parsing (that step probably happens anyway during the `!=` operation, even if it is a bit hidden. Or just call it for the printing.

aleju replied:
Added `np.dtype(dtype)` to the `format()` command in the latest commit. Example output now: `TypeError: Supplied output array has the wrong type. Expected float64, got float32`

aleju committed:
MAINT: Improve error message dtype appearance

This changes the string conversion of an expected
dtype in an error message from e.g.
"<class 'numpy.float64'>" to "float64".

charris commented:
Thanks @aleju .
